app/controllers/admin/statistics_announcement_publications_controller.rb

```diff
@@ -0,0 +1,61 @@
+class Admin::StatisticsAnnouncementPublicationsController < Admin::BaseController
+  before_action :find_statistics_announcement
+  layout "design_system"
+
+  def index
+    if params[:search].present?
+      filter
+    end
+  end
+
+  def connect
+    @statistics_announcement.assign_attributes(publication_params)
+
+    if @statistics_announcement.save
+      redirect_to [:admin, @statistics_announcement], notice: "Announcement updated successfully"
+    else
+      filter
+      render :index
+    end
+  end
+
+private
+
+  def filter
+    @filter ||= Admin::EditionFilter.new(get_editions, current_user, edition_filter_options)
+  end
+
+  def get_editions
+    Edition.statistical_publications.with_title_containing(params[:search])
```

Comment:
hey chris, great work. I noticed one thing here that i think is causing a bit of an issue. Often when i try and link the document i'm hitting this error


<img width="1034" alt="image" src="https://github.com/alphagov/whitehall/assets/42515961/d81a37c0-6be3-4761-926e-efa55fb266cd">

Looking at a statistics announcement i can see that it can either be linked a publication with the subtype of official on national statistics.

<img width="752" alt="image" src="https://github.com/alphagov/whitehall/assets/42515961/27b8522e-7781-42a0-a65b-306e5a6bbfbb">

Looking at the JS implementation i can see that the publication_type_id is passed into the JS module

<img width="946" alt="image" src="https://github.com/alphagov/whitehall/assets/42515961/e485df2c-ee7d-4a06-afd8-6636aa781bf4">

This is used to only return publications with a title of that subtype when the ajax call is made

<img width="619" alt="image" src="https://github.com/alphagov/whitehall/assets/42515961/f68fbb00-36f4-4c9c-a456-715bc8bae721">

<img width="664" alt="image" src="https://github.com/alphagov/whitehall/assets/42515961/48cc5091-970f-422c-8d2b-801263ee3b78">

I think we need to update this method to only pass in editions with the statistics announcements `publication_type_id`

As you've set this up in a before_action we can just do
```
    Edition.statistical_publications.where(publication_type_id: @statistics_announcement.publication_type_id).with_title_containing(params[:search])

```
When i pass this extra where clause it reduces the amount of returned editions from 250 to 20. All of which can be linked 

<img width="651" alt="image" src="https://github.com/alphagov/whitehall/assets/42515961/fad06813-2277-429d-92f2-427a0ed0e5b6">


<img width="409" alt="image" src="https://github.com/alphagov/whitehall/assets/42515961/80cff813-844a-4ad5-9667-b2fa9596cf48">